langchain/vectorstores/faiss.py

```diff
@@ -512,7 +512,9 @@ def __from(
             metadata = metadatas[i] if metadatas else {}
             documents.append(Document(page_content=text, metadata=metadata))
         index_to_id = dict(enumerate(ids))
-        docstore = InMemoryDocstore(dict(zip(index_to_id.values(), documents)))
+        docstore = InMemoryDocstore(
+            {index_to_id[i]: doc for i, doc in enumerate(documents)}
```

Comment:
could this just be `dict(zip(ids, documents))` instead?

Reply:
that is what I have mentioned in the description. If you pass two lists to zip it returns a tuple of length smallest of the two. if `ids = ['A']` and `texts=["I love Open Source","I love langchain"]` then only one docstore will be created with id value 'A'.

Reply:
imo we should just check that the two lists are the same. if they are not, we can raise a really clear error. even this error here (key error) is not that clear what is happening

Reply:
Agree. I think a similar change will be needed in annoy.py as well.